### PR TITLE
Switch to nlohmann/json

### PR DESCRIPTION
**Pull request by wgenh**
**Description**
For https://github.com/commaai/openpilot/issues/24523
Switched code to nlohmann/json and modified tinygrad's code to encode binary data into base64 and use utf-8

**Verification**

USE_THNEED=1 selfdrive/modeld/modeld and selfdrive/modeld/tests/test_models.py now work

### CODE REVIEW

File: SConstruct [hunk 1]
```diff
@@ -202,7 +202,8 @@ env = Environment(
     "#third_party/acados/include/hpipm/include",
     "#third_party/catch2/include",
     "#third_party/libyuv/include",
-    "#third_party/json11",
+    "#third_party/nlohmann",
+    "#third_party/tobiaslocker",
```
Comment (adeebshihadeh) [inline SConstruct:206]: we're very careful about adding new dependencies. can you find some other way to do this?

**Review by adeebshihadeh — CHANGES_REQUESTED** [review]

File: SConstruct (hunk 1 above)
Comment (wgenh) [reply SConstruct:206]: I can implement base64 decoding without an external library.

Comment (wgenh) [reply SConstruct]: Done.

**Comment by wgenh** [comment]
How can I add changes to the tinygrad code?

**Comment by wgenh** [comment]
There is no input validation in my code. It should probably be okay.

**Comment by wgenh** [comment]
Should I mark this as ready for review?

**Comment by wgenh** [comment]
The pre-commit fails because json.hpp isn't in lfs, right?

**Comment by wgenh** [comment]
Sorry to bother, but can anyone of you check this, please?
@sshane @adeebshihadeh @jnewb1

**Comment by wgenh** [comment]
Merged latest changes to logreader, that should fix unit tests.

**Comment by wgenh** [comment]
How can I merge changes to the tinygrad?

**Pull request closed without merging**